carbonmark-api/src/utils/helpers/users.utils.ts

```diff
@@ -12,21 +12,29 @@ type FbUserDocument = {
 };
 /**
  * This function retrieves a user by their wallet address from the Firestore database.
- * @param {app.App} fb - The Firebase app instance.
- * @param {string} wallet - The wallet address of the user.
- * @returns {Promise<any>} The data of the user if found, otherwise null.
  */
-export const getUserDocumentByWallet = async (
-  fb: app.App,
-  wallet: string
-): Promise<FirebaseFirestore.DocumentData | undefined> =>
-  await fb.firestore().collection("users").doc(wallet.toUpperCase()).get();
+export const getProfileByAddress = async (params: {
+  firebase: app.App;
+  address: string;
+}): Promise<any> => {
+  const doc = await params.firebase
+    .firestore()
+    .collection("users")
+    .doc(params.address.toUpperCase())
+    .get();
+  if (!doc.exists) return null;
+  return doc.data();
+};
 
-export const getUserByWallet = async (
-  fb: app.App,
-  wallet: string
-): Promise<FirebaseFirestore.DocumentData | undefined> =>
-  (await getUserDocumentByWallet(fb, wallet))?.data();
+interface UserProfile {
```

Comment:
There is an `FbUserDocument` type in this file. Can you replace that with this one or vice-versa?

Reply:
missed that after merging, thanks!